xarray/core/missing.py

```diff
@@ -0,0 +1,258 @@
+
+    if dim is None:
+        raise NotImplementedError('dim is a required argument')
+
+    method, index = get_clean_interp_args(arr, method, dim, **kwargs)
+    kwargs.update(kind=method)
+
+    interpolator = _get_interpolator(method, **kwargs)
+
+    return apply_ufunc(wrap_interpolator,
+                       interpolator, index, arr, kwargs=kwargs,
+                       input_core_dims=[[], [dim], [dim]],
+                       output_core_dims=[[dim]],
+                       dask='forbidden',
+                       vectorize=True,
+                       keep_attrs=True).transpose(*arr.dims)
+
+
+def wrap_interpolator(interpolator, x, y, **kwargs):
+    '''helper function to apply interpolation along 1 dimension'''
+    if x.shape != y.shape:
+        # this can probably be removed once I get apply_ufuncs to work
+        raise AssertionError("x and y shapes do not match "
+                             "%s != %s" % (x.shape, y.shape))
+
+    nans = pd.isnull(y)
+    nonans = ~nans
+
+    # fast track for no-nans and all-nans cases
+    n_nans = nans.sum()
+    if n_nans == 0 or n_nans == len(y):
+        return y
+
+    f = interpolator(x[nonans], y[nonans], **kwargs)
+    return f(x[nans])
+
+
+def _bfill(arr, n=None, axis=-1):
+    '''inverse of ffill'''
+    import bottleneck as bn
+
+    arr = np.flip(arr, axis=axis)
+
+    # fill
+    arr = bn.push(arr, axis=axis, n=n)
+
+    # reverse back to original
+    return np.flip(arr, axis=axis)
+
+
+def ffill(self, dim=None, limit=None):
+    ''' '''
+    import bottleneck as bn
+
+    return apply_ufunc(bn.push, self,
+                       dask='forbidden',
+                       keep_attrs=True,
+                       kwargs=dict(n=limit, axis=self.get_axis_num(dim)))
+
+
+def bfill(self, dim=None, limit=None):
+    ''' '''
+    return apply_ufunc(_bfill, self,
+                       dask='forbidden',
+                       keep_attrs=True,
+                       kwargs=dict(n=limit, axis=self.get_axis_num(dim)))
+
+
+def _get_interpolator(method, **kwargs):
+    interp1d_methods = ['linear', 'nearest', 'zero', 'slinear', 'quadratic',
+                        'cubic', 'polynomial']
+
+    if (method == 'linear' and not
+            kwargs.get('fill_value', None) == 'extrapolate'):
+        return NumpyInterpolator
+    else:
+        try:
+            from scipy import interpolate
+            if method in interp1d_methods:
+                return ScipyInterpolator
+            elif method == 'piecewise_polynomial':
+                return FromDerivativesInterpolator
+            elif method == 'barycentric':
+                return interpolate.BarycentricInterpolator
+            elif method == 'krog':
+                return interpolate.KroghInterpolator
+            elif method == 'pchip':
+                return interpolate.PchipInterpolator
+            elif method == 'spline':
+                return SplineInterpolator
+        except ImportError:
+            pass
```

Comment:
It would probably be more useful to raise an error about how scipy is needed for the given interpolator rather than a generic error message.